server/main/src/main.rs

```diff
@@ -58,53 +58,61 @@ mod url_norm;
 #[cfg(test)]
 mod test;
 
+pub fn is_top_level(path: &Path) -> bool {
+    let path = path.as_os_str().to_str().unwrap().replace("\\", "/");
```

Comment:
Can probably replace this with:
```suggestion
    use path_slash::PathExt;
    let path = path.to_slash().unwrap();
```